store/snap_remote_repo.go

```diff
@@ -232,6 +257,111 @@ func (s *SnapUbuntuStoreRepository) checkStoreResponse(resp *http.Response) {
 	}
 }
 
+func getRequiresPurchase(prices map[string]float64, purchases purchaseList) bool {
+	// if the snap is free, then it doesn't need purchasing
+	if len(prices) == 0 {
+		return false
+	}
+
+	for _, purchase := range purchases {
+		// if the purchase is not an in-app purchase
+		if purchase.ItemSKU == "" {
+			return false
+		}
+	}
+
+	// the snap is not free, and we couldn't find a purchase
```

Comment:
Where is the "found a purchase" logic?  Loop above has a single condition which is documented to be about in-app purchases rather than previously made purchases.